gr-digital/lib/symbol_sync_cc_impl.cc

```diff
@@ -132,6 +129,48 @@ symbol_sync_cc_impl::symbol_sync_cc_impl(enum ted_type detector_type,
     set_output_multiple(d_osps_n);
 }
 
+void symbol_sync_cc_impl::set_sps(float sps)
+{
+    gr::thread::scoped_lock l(d_setlock);
+
+    d_sps = sps;
+
+    const auto max_period = sps + d_max_deviation;
+    const auto min_period = sps - d_max_deviation;
+    d_clock.set_max_avg_period(max_period);
+    d_clock.set_min_avg_period(min_period);
+    d_clock.set_nom_avg_period(sps);
+    d_clock.set_avg_period(sps);
+    d_clock.set_inst_period(sps);
+    d_clock.set_phase(0.0f);
+
+    d_inst_output_period = sps / d_osps;
+    d_inst_clock_period = sps;
+    d_avg_clock_period = sps;
+
+    sync_reset_internal_clocks();
+
+    d_inst_interp_period = d_inst_clock_period / d_interps_per_symbol;
+
+    check_interps();
```

Comment:
Do we want this in the regular ctor, too?

Reply:
Line 114 ... or are you saying it shouldn't be in the constructor?

Reply:
The code that is now in `check_interps()` (which didn't exist before) was originally only in the constructor. I wanted to do the same checks in the constructor and in the `set_sps()` method, so I extracted them to a common function.